typegen/src/typegen/settings/derives.rs

```diff
@@ -136,3 +136,161 @@ impl ToTokens for Derives {
         }
     }
 }
+
+/// This is like a DerivesRegistry, but the recursive type derives have been flattened out into specific_type_derives.
+///
+/// Can be constructed properly using a DerivesRegistry and a PortableRegistry with `DerivesRegistry::flatten_recursive_derives()`.
+#[derive(Debug, Clone, Default)]
+pub struct FlatDerivesRegistry {
+    default_derives: Derives,
+    specific_type_derives: HashMap<syn::TypePath, Derives>,
+}
+
+impl FlatDerivesRegistry {
+    /// Resolve the derives for a specific type path.
+    pub fn resolve(&self, ty: &syn::TypePath) -> Derives {
+        let mut resolved_derives = self.default_derives.clone();
+        if let Some(specific) = self.specific_type_derives.get(ty) {
+            resolved_derives.extend_from(specific.clone());
+        }
+        resolved_derives
+    }
+
+    /// Resolve the derives for a specific type.
+    pub fn resolve_derives_for_type(
+        &self,
+        ty: &Type<PortableForm>,
+    ) -> Result<Derives, TypegenError> {
+        Ok(self.resolve(&syn_type_path(ty)?))
+    }
+}
+
+impl DerivesRegistry {
```

Comment:
Nit; merge this with the above impl block so all impls in one place below the `DerivesRegistry` definition?